core/dbt/clients/system.py

```diff
@@ -407,6 +407,11 @@ def run_cmd(
     logger.debug('Executing "{}"'.format(' '.join(cmd)))
     if len(cmd) == 0:
         raise dbt.exceptions.CommandError(cwd, cmd)
+    exe_pth = shutil.which(cmd[0])
+    if not exe_pth:
+        # executable not found
+        raise dbt.exceptions.CommandError(cwd, cmd)
```

Comment:
I think we want to raise an `ExecutableError` here instead (like @jtcohen6 mentioned) there are some places where we present relevant error messages for an `ExecutableError` with git commands